addons/promote/jaxrs/src/main/java/org/commonjava/indy/promote/bind/jaxrs/PromoteAdminResource.java

```diff
@@ -0,0 +1,189 @@
+/**
+ * Copyright (C) 2011-2020 Red Hat, Inc. (https://github.com/Commonjava/indy)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.commonjava.indy.promote.bind.jaxrs;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiImplicitParam;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import org.commonjava.indy.bind.jaxrs.IndyResources;
+import org.commonjava.indy.bind.jaxrs.SecurityManager;
+import org.commonjava.indy.bind.jaxrs.util.REST;
+import org.commonjava.indy.bind.jaxrs.util.ResponseHelper;
+import org.commonjava.indy.model.core.StoreKey;
+import org.commonjava.indy.promote.data.PromotionException;
+import org.commonjava.indy.promote.model.GroupPromoteResult;
+import org.commonjava.indy.promote.model.ValidationRuleDTO;
+import org.commonjava.indy.promote.model.ValidationRuleSet;
+import org.commonjava.indy.promote.validate.PromoteValidationsManager;
+import org.commonjava.indy.util.ApplicationContent;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.SecurityContext;
+import javax.ws.rs.core.UriInfo;
+
+import static org.commonjava.indy.util.ApplicationContent.application_json;
+
+@Api( value = "Promote administration resource to manage configurations for content promotion." )
+@Path( "/api/admin/promotion" )
+@Produces( { application_json } )
+@REST
+public class PromoteAdminResource
+        implements IndyResources
+{
+
+    private final Logger logger = LoggerFactory.getLogger( getClass() );
+
+    @Inject
+    private PromoteValidationsManager validationsManager;
+
+    @Inject
+    private ObjectMapper mapper;
+
+    @Inject
+    private SecurityManager securityManager;
+
+    @Inject
+    private ResponseHelper responseHelper;
+
+    @ApiOperation( "Promote a source repository into the membership of a target group (subject to validation)." )
+    @ApiResponse( code = 200, message = "Promotion operation finished (consult response content for success/failure).",
```

Comment:
Because of indepotency of "PUT" http requests i think it is better to return 201 - 204 status codes based on type of response returned

Reply:
Oops, copy-paste error. Will fix in next commit.